### PR TITLE
[AEROGEAR-2734] List mobile builds in client overview

### PR DESCRIPTION
**Pull request by dimitraz**
**Description**
List the latest 5 builds of the mobile client
The build link should point to its pipeline build view

**Verification Steps**
- Provision an android mobile client and the mobile ci/cd service
- Create several new pipelines and builds and verify that they are correctly showing up in the mobile client row
- Verify that there is only one build per pipeline in the mobile client row 
- Verify that the pipelines are listed in order of latest builds
- Verify that no more than 5 pipelines are displayed in the mobile client row

**Related Jira**
[AEROGEAR-2734](https://issues.jboss.org/browse/AEROGEAR-2734)

### CODE REVIEW

**Comment by dimitraz** [comment]
@JameelB

File: app/scripts/directives/overview/mobileClientRow.js [hunk 1]
```diff
@@ -52,6 +54,26 @@
...
+          return _.get(build, 'metadata.labels.mobile-client-id') === _.get(row, 'apiObject.metadata.name');
+        });
+
+        // Group all builds by buildconfig name
+        row.builds = _.groupBy(row.builds, function(build) {
+          return _.get(build, 'metadata.labels.buildconfig');
+        });
+
+        // Return latest build from each group
+        row.builds = _.map(row.builds, function(buildArray) {
```
Comment (ghost) [inline app/scripts/directives/overview/mobileClientRow.js]: same here, assign result to new variable

File: app/scripts/directives/overview/mobileClientRow.js [hunk 2]
```diff
@@ -52,6 +54,26 @@
...
+          return _.last(buildArray);
+        });
+
+        // Get the latest 5 builds
+        row.builds = _.slice(orderObjectsByDate(row.builds, true), 0, 5);
```
Comment (ghost) [inline app/scripts/directives/overview/mobileClientRow.js]: same here, assign result to new variable

File: app/scripts/directives/overview/mobileClientRow.js [hunk 3]
```diff
@@ -52,6 +54,26 @@
...
       });
+
+      DataService.watch(buildsVersion, row.context, function(buildsData) {    
+        // Filter builds by mobile client id
+        row.builds = _.filter(buildsData.by('metadata.name'), function(build) {
+          return _.get(build, 'metadata.labels.mobile-client-id') === _.get(row, 'apiObject.metadata.name');
+        });
+
+        // Group all builds by buildconfig name
+        row.builds = _.groupBy(row.builds, function(build) {
```
Comment (ghost) [inline app/scripts/directives/overview/mobileClientRow.js]: Do not assign to same variable or create copy of `row.builds`

File: app/scripts/directives/overview/mobileClientRow.js [hunk 4]
```diff
@@ -61,7 +83,14 @@
       var opts = {
         tab: "mobileServices"
       };
-      $location.url(Navigate.resourceURL(resource, kind, namespace, null, opts));
+      return Navigate.resourceURL(resource, kind, namespace, null, opts);
+    }
+
+    row.navigateToBuild = function(build) {
+      var resource = _.get(build, 'metadata.labels.buildconfig');
+      var kind = 'BuildConfig';
```
Comment (ghost) [inline app/scripts/directives/overview/mobileClientRow.js]: `const KIND = 'BuildConfig';` ?

**Review by vchepeli — CHANGES_REQUESTED** [review]

File: app/scripts/directives/overview/mobileClientRow.js [hunk 5]
```diff
@@ -52,6 +55,26 @@
...
+          return _.get(build, 'metadata.labels.buildconfig');
+        });
+
+        // Return latest build from each group
+        var builds = _.map(groupedBuilds, function(buildArray) {
+          return _.last(buildArray);
+        });
+
+        // Get the latest 5 builds
+        row.builds = _.slice(orderObjectsByDate(builds, true), 0, 5);
```
Comment (sedroche) [inline app/scripts/directives/overview/mobileClientRow.js]: I prefer to keep filters within the html as much as possible. There is a `limitTo` filter which can be used along with `orderObjectsByDate`

File: app/views/overview/_mobile-client-row.html [hunk 6]
```diff
@@ -57,13 +57,32 @@ <h3>
...
+            <div class="component-label section-label">Mobile Builds</div>
+            <div ng-repeat="build in row.builds">
+              <div class="row">
+                <div class="col-md-6">
+                  <h3>
+                    <a href="" ng-href="{{ row.navigateToBuild(build) }}">{{ build.metadata.labels.buildconfig }}</a>
+                  </h3>
+                </div>
+                <div class="col-md-6">
+                    <span class="build-status-icon" ng-if="build.status.phase.toLowerCase() === 'running'"><i class="fa fa-refresh"></i></span>
```
Comment (sedroche) [inline app/views/overview/_mobile-client-row.html]: Can we use the exact html used in the pipelines page I think its in `views/directives/_build-pipeline-expanded.html`

File: app/scripts/directives/overview/mobileClientRow.js [hunk 7]
```diff
@@ -52,6 +55,26 @@
...
       });
+
+      DataService.watch(buildsVersion, row.context, function(buildsData) {    
+        // Filter builds by mobile client id
+        var filteredBuilds = _.filter(buildsData.by('metadata.name'), function(build) {
+          return _.get(build, 'metadata.labels.mobile-client-id') === _.get(row, 'apiObject.metadata.name');
+        });
+
+        // Group all builds by buildconfig name
+        var groupedBuilds = _.groupBy(filteredBuilds, function(build) {
```
Comment (sedroche) [inline app/scripts/directives/overview/mobileClientRow.js]: I believe we are showing the last five builds so we aren't grouping by buildconfig

**Review by vchepeli — APPROVED** [review]

**Comment by dimitraz** [comment]
@sedroche @vchepeli rebased, can you take another look?

File: app/views/overview/_mobile-client-row.html [hunk 8]
```diff
@@ -58,24 +58,54 @@ <h3>
               </button>
               <span class="pficon pficon-info"></span>
               {{row.servicesNotBoundCount}} mobile services are not bound to this client.
-              <a href="" ng-click="row.navigateToMobileServices()">Bind them to use with this client.</a>
+              <a href="" ng-href="{{ row.navigateToMobileServices() }}">Bind them to use with this client.</a>
```
Comment (JameelB) [inline app/views/overview/_mobile-client-row.html]: This needs to be changed to `row.navigateToMobileTab()`

**Review by JameelB — APPROVED** [review]

**Pull request merged**